keyboards/preonic/keymaps/alfrdmalr/keymap.c

```diff
@@ -0,0 +1,271 @@
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |      |      |      |      |      |      |      |      |      |      |  Del |
+ * |------+------+------+------+------+-------------+------+------+------+------+------|
+ * |      |      |      |Aud on|AudOff|AGnorm|AGswap|Qwerty|Colemk|Dvorak|      |      |
+ * |------+------+------+------+------+------|------+------+------+------+------+------|
+ * | ASTG |Voice-|Voice+|Mus on|MusOff|MidiOn|MidOff|      |      |      |      |      |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |      |      |      |      |             |      |      |      |      |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_SETTINGS] = LAYOUT_preonic_grid( \
+  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  \
+  _______, _______,   DEBUG,   _______, _______, _______, _______, TERM_ON, TERM_OFF,_______, _______, KC_DEL,  \
+  _______, _______, MU_MOD,  AU_ON,   AU_OFF,  AG_NORM, AG_SWAP, QWERTY,  _______, _______,  _______, _______, \
+  KC_ASTG, MUV_DE,  MUV_IN,  MU_ON,   MU_OFF,  MI_ON,   MI_OFF,  _______, _______, _______, _______, _______, \
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______  \
+)
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+        case QWERTY:
+          if (record->event.pressed) {
+            set_single_persistent_default_layer(_QWERTY);
+          }
+          return false;
+          break;
+        // case COLEMAK:
+        //   if (record->event.pressed) {
+        //     set_single_persistent_default_layer(_COLEMAK);
+        //   }
+        //   return false;
+        //   break;
+        // case DVORAK:
+        //   if (record->event.pressed) {
+        //     set_single_persistent_default_layer(_DVORAK);
+        //   }
+        //   return false;
+        //   break;
+        case SYMBOLS:
+          if (record->event.pressed) {
+            layer_on(_SYMBOLS);
+          } else {
+            layer_off(_SYMBOLS);
+          }
+          return false;
+          break;
+        case NUMPAD:
+          if (record->event.pressed) {
+            layer_on(_NUMPAD);
+          } else {
+            layer_off(_NUMPAD);
+          }
+          return false;
+          break;
+        case NAVIGATION:
+          if (record->event.pressed) {
+            layer_on(_NAVIGATION);
+          } else {
+            layer_off(_NAVIGATION);
+          } 
+          return false;
+          break;
+        case SETTINGS:
+          if (record->event.pressed) {
+            layer_on(_SETTINGS);
+          } else {
+            layer_off(_SETTINGS);
+          } 
+          return false;
+          break;
```

Comment:
You don't need any of these. 

`MO(xxx)` will do exactly this. No need to make a custom macro for this.

Reply:
DUH. Thanks for that - I had originally planned on using a trilayer, but decided against it and evidently never got rid of my boilerplate for it.